### PR TITLE
let svgo handle dataURL

### PR DESCRIPTION
**Pull request by holymonson**
Directly call decodeURIComponent(string) may throw 'URIError: URI malformed' if the
string is unencoded but contains '%'. Just let svgo handle the pre-check and
decode/encode.

### CODE REVIEW

**Comment by TrySound** [comment]
@holymonson Fix test case please

**Comment by holymonson** [comment]
@TrySound Review please

File: src/lib/url.js [hunk 1]
```diff
@@ -1,12 +0,0 @@
...
-        .replace(/"/g,   '\'')
-        .replace(/%/g,   '%25')
-        .replace(/</g,   '%3C')
-        .replace(/>/g,   '%3E')
-        .replace(/&/g,   '%26')
-        .replace(/#/g,   '%23')
-        .replace(/\s+/g, ' ');
-};
-
-export let decode = decodeURIComponent;
```
Comment (TrySound) [inline src/lib/url.js]: This is mininal encoding required in browsers. svgo uses full encoding, which not optimal.

File: src/__tests__/index.js [hunk 2]
```diff
@@ -3,41 +3,39 @@ import postcss from 'postcss';
...
-import {encode, decode} from '../lib/url';
-import {readFileSync as file} from 'fs';
 
 let name = pkg.name;
 
 let tests = [{
     message: 'should optimise inline svg',
-    fixture: 'h1{background:url(\'data:image/svg+xml;utf-8,<?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve"><circle cx="50" cy="50" r="40" fill="yellow" /><!--test comment--></svg>\')}',
-    expected: 'h1{background:url(\'data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg"><circle cx="50" cy="50" r="40" fill="#ff0"/></svg>\')}',
+    fixture: 'h1{background:url(\'data:image/svg+xml,<?xml version="1.0" encoding="utf-8"?><!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 1.1//EN" "http://www.w3.org/Graphics/SVG/1.1/DTD/svg11.dtd"><svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xml:space="preserve"><circle cx="50" cy="50" r="40" fill="yellow" /><!--test comment--></svg>\')}',
```
Comment (TrySound) [inline src/__tests__/index.js]: Why did you removed encoding? It's required by IE.
http://stackoverflow.com/questions/32870788/css-using-raw-svg-in-the-url-parameter-of-a-background-image-in-ie/32871853#32871853

**Comment by ben-eb** [comment]
https://github.com/ben-eb/postcss-svgo/commit/a48c650c4b3ed504f72195e8dc4bae6f3666faca

**Pull request closed without merging**